tools/run-bidi-server.mjs

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env node
+
+/**
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {spawn, spawnSync} from 'child_process';
+import {createWriteStream, mkdirSync} from 'fs';
+import {basename, join} from 'path';
+import {packageDirectorySync} from 'pkg-dir';
+
+import yargs from 'yargs';
+import {hideBin} from 'yargs/helpers';
+
+// Changing the current work directory to the package directory.
+process.chdir(packageDirectorySync());
+
+function log(message) {
+  // eslint-disable-next-line no-console
+  console.log(`(${basename(process.argv[1])}) ${message}`);
+}
+
+const argv = yargs(hideBin(process.argv))
+  .usage(
+    `[CHANNEL=<stable | beta | canary | dev>] [DEBUG=*] [DEBUG_COLORS=<yes | no>] [HEADLESS=<true | false>] [LOG_DIR=logs] [NODE_OPTIONS=--unhandled-rejections=strict] [PORT=8080] $0`
+  )
+  .option('headless', {
+    describe:
+      'Whether to start the server in headless or headful mode. The --headless flag takes precedence over the HEADLESS environment variable.',
+    type: 'boolean',
+    default: process.env.HEADLESS === 'true',
+  })
+  .parse();
+
+let BROWSER_BIN = process.env.BROWSER_BIN;
+let CHANNEL = process.env.CHANNEL || 'local';
+if (CHANNEL === 'local') {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  BROWSER_BIN = spawnSync('node', ['tools/install-browser.mjs', '--shell'])
+    .stdout.toString()
+    .trim();
+  // Need to pass valid CHANNEL to the command below
+  CHANNEL = 'canary';
+}
+
+const DEBUG = process.env.DEBUG ?? '*';
```

Comment:
Maybe preserve the comment from the bash script here, otherwise there's the risk future refactoring will break it.

```
// DEBUG= (empty string) is allowed.
```

Reply:
why do we need an empty string?

Reply:
DEBUG is set to '*' by default. If you do not want any DEBUG, you set it to empty.